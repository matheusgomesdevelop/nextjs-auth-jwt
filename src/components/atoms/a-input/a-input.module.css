.a_input {
    width: 100%;
    height: 48px;
    padding: var(--spacing-12) var(--spacing-16);
    font-size: var(--fs-text-md);
    color: var(--text-color);
    background-color: var(--white);
    border: 1px solid var(--gray-300);
    border-radius: 8px;
    transition: background-color 0.2s, color 0.2s, border-color 0.2s, box-shadow 0.2s;
    box-shadow: 0px 1px 2px 0px rgba(16, 24, 40, 0.05);
}

.a_input::placeholder {
    color: var(--gray-500);
    font-size: var(--fs-text-md);
    font-weight: var(--fw-regular);
}