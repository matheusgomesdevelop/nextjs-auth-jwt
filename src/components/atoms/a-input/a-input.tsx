import { InputTextProps, InputText } from "primereact/inputtext";

import styles from "./a-input.module.css";

export const A_Input = ({ ...props }: InputTextProps) => {
  return (
    <InputText
      data-testid="a-input"
      pt={{
        root: {
          className: styles.a_input,
        },
      }}
      {...props}
    />
  );
};
