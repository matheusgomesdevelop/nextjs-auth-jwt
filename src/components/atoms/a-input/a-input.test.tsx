import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

// Atoms
import { AInput } from "./a-input";

describe("Deve renderizar o input corretamente", () => {
  it("Deve renderizar o input com o type", () => {
    // Arrange
    render(<AInput type="email" name="" placeholder="" />);

    // Act
    const get_input = screen.getByTestId("a-input");

    // Assert
    expect(get_input).toBeInTheDocument();
    expect(get_input).toHaveAttribute("type");
    expect(get_input.getAttribute("type")).not.toHaveLength(0);
});
it("Deve renderizar o input com o name", () => {
    // Arrange
    render(<AInput type="" name="email" placeholder="" />);

    // Act
    const get_input = screen.getByTestId("a-input");

    // Assert
    expect(get_input).toBeInTheDocument();
    expect(get_input).toHaveAttribute("name");
    expect(get_input.getAttribute("name")).not.toHaveLength(0);
});
it("Deve renderizar o input com o placeholder", () => {
    // Arrange
    const placeholder_text = "Placeholder e-mail";
    render(<AInput type="" name="" placeholder={placeholder_text} />);

    // Act
    const get_input = screen.getByTestId("a-input");

    // Assert
    expect(get_input).toBeInTheDocument();
    expect(get_input).toHaveAttribute("placeholder");
    expect(get_input.getAttribute("placeholder")).not.toHaveLength(0);
});
});
