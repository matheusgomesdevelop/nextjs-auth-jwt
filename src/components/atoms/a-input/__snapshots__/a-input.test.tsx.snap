// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar o a-input corretamente. 1`] = `
<input
  className="p-inputtext p-component a_input"
  data-pc-name="inputtext"
  data-pc-section="root"
  data-testid="a-input"
  onBeforeInput={[Function]}
  onInput={[Function]}
  onKeyDown={[Function]}
  onPaste={[Function]}
/>
`;
